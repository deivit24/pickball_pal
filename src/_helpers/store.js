import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
