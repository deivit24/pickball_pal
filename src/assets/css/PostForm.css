#PostForm {
  margin-bottom: 25px;
}
#PostForm form {
  width: 360px;
  margin: 0px auto;
}
/* #PostForm button {
  background-color: white;
  color: #3f50b5;
  font-weight: 700;
  margin: 10px auto;
}
#PostForm button:hover {
  background-color: #3f50b5;
  color: white;
  border: 1px solid white;
  font-weight: 700;
  margin: 10px auto;
} */

#PostForm label {
  color: black;
}
#PostForm textarea {
  min-width: 270px;
  max-width: 280px;
}
