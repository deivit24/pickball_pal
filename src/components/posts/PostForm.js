import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { addPost } from '../../_actions/post';
import Input from '@material-ui/core/Input';
import Button from '@material-ui/core/Button';
import '../../assets/css/PostForm.css';
const PostForm = () => {
  const dispatch = useDispatch();
  const [text, setText] = useState('');

  const handleChange = (e) => {
    setText(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(addPost({ text }));
    setText('');
  };
  return (
    <div id="PostForm">
      <form action="" autoComplete="off" noValidate>
        <Input
          id="outlined-multiline-flexible"
          label="New Post"
          multiline
          className="mr-2"
          rowsMax={4}
          name="text"
          value={text}
          onChange={handleChange}
          variant="filled"
          placeholder="New Post"
        />

        <Button
          onClick={handleSubmit}
          size="small"
          variant="contained"
          color="primary"
        >
          Post
        </Button>
      </form>
    </div>
  );
};

export default PostForm;
