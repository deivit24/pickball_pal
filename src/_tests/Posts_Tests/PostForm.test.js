import React from 'react';
import { render } from '@testing-library/react';
import PostForm from '../../components/posts/PostForm';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing PostForm

describe('Testing the PostForm', () => {
  it('it renders PostForm without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <PostForm />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <PostForm />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
