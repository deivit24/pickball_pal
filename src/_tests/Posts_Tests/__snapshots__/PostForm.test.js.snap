// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the PostForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="PostForm"
  >
    <form
      action=""
      autocomplete="off"
      novalidate=""
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline mr-2 MuiInputBase-multiline MuiInput-multiline"
        label="New Post"
        variant="filled"
      >
        <textarea
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          id="outlined-multiline-flexible"
          name="text"
          placeholder="New Post"
          rows="1"
          style="height: 0px; overflow: hidden;"
        />
        <textarea
          aria-hidden="true"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
          tabindex="-1"
        />
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Post
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;
